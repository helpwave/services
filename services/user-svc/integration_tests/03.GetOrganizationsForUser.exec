#!/bin/bash

echo "Exists, has members:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiI4YjVmZjkwYS03NTQ4LTQ1NGUtOTY4NC02ZjJjMzA2NzdkYjUiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{}' \
	localhost:3002 \
	proto.services.user_svc.v1.OrganizationService/GetOrganizationsForUser 2>&1 || echo ""

echo "Does not exist:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{}' \
	localhost:3002 \
	proto.services.user_svc.v1.OrganizationService/GetOrganizationsForUser 2>&1 || echo ""
