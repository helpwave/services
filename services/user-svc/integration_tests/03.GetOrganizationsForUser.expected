Exists, has members:
{
  "organizations": [
    {
      "id": "ead68d1d-db6b-4cf0-b316-ac9064c66413",
      "longName": "Uniklinik Aachen",
      "shortName": "Aachen",
      "contactEmail": "testine.test@helpwave.de",
      "members": [
        {
          "userId": "51dce0ea-2e72-412a-988e-8e9089a68572",
          "email": "maike.test@helpwave.de",
          "nickname": "Maike",
          "avatarUrl": "https://source.boringavatars.com/marble/128/"
        },
        {
          "userId": "8b5ff90a-7548-454e-9684-6f2c30677db5",
          "email": "heinrich.heine@helpwave.de",
          "nickname": "Heine",
          "avatarUrl": "https://source.boringavatars.com/marble/128/"
        }
      ]
    }
  ]
}
Does not exist:
{}
